docs/content/concepts/dagster-pipes/aws-emr-serverless.mdx

```diff
@@ -0,0 +1,158 @@
+---
+title: "Integrating AWS EMR Serverless with Dagster Pipes | Dagster Docs"
+description: "Learn to integrate Dagster Pipes with AWS EMR Serverless to launch external code from Dagster assets."
+---
+
+# AWS EMR Serverless & Dagster Pipes
+
+This tutorial gives a short overview on how to use [Dagster Pipes](/concepts/dagster-pipes) with [AWS EMR Serverless](https://aws.amazon.com/emr-serverless/).
+
+The [dagster-aws](/\_apidocs/libraries/dagster-aws) integration library provides the <PyObject object="PipesEMRServerlessClient" module="dagster_aws.pipes" /> resource which can be used to launch AWS EMR Serverless jobs from Dagster assets and ops. Dagster can receive regular events like logs, asset checks, or asset materializations from jobs launched with this client. Using it requires minimal code changes on the job side.
+
+---
+
+## Prerequisites
+
+- **In the orchestration environment**, you'll need to:
+
+  - Install the following packages:
+
+    ```shell
+    pip install dagster dagster-webserver dagster-aws
+    ```
+
+    Refer to the [Dagster installation guide](/getting-started/install) for more info.
+
+  - **AWS authentication credentials configured.** If you don't have this set up already, refer to the [boto3 quickstart](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/quickstart.html).
+
+- **In AWS**:
+
+  - An existing AWS account
+  - An AWS EMR Serverless job. AWS CloudWatch logging has to be enabled in order to receive logs from the job:
+
+  ```json
+  {
+    "monitoringConfiguration": {
+      "cloudWatchLoggingConfiguration": { "enabled": true }
+    }
+  }
+  ```
+
+---
+
+## Step 1: Install the dagster-pipes module
+
+There are a [few options](https://docs.aws.amazon.com/emr/latest/EMR-Serverless-UserGuide/using-python-libraries.html) available for shipping Python packages to a PySpark job. For example, [install it in your Docker image](https://docs.aws.amazon.com/emr/latest/EMR-Serverless-UserGuide/application-custom-image.html):
+
+Install the `dagster-pipes` module in the image used for your EMR job. For example, you can install the dependency with `pip` in your image Dockerfile:
+
+```Dockerfile
+# start from EMR image
+FROM public.ecr.aws/emr-serverless/spark/emr-7.2.0:latest
+
+USER root
+
+RUN python -m pip install dagster-pipes
+
+# copy the jon script
+COPY . .
+
+USER hadoop
+```
+
+---
+
+## Step 2: Add dagster-pipes to the EMR Serverless job script
+
+Call `open_dagster_pipes` in the EMR Serverless script to create a context that can be used to send messages to Dagster:
```

Comment:
is there a hello world for EMR Serverless scripts that we can use here too? 
maybe adding
```
from pyspark.sql import SparkSession
from dagster_pipes import (
    PipesEnvVarParamsLoader,
    PipesS3ContextLoader,
    open_dagster_pipes,
)

def main():
    with open_dagster_pipes() as pipes:
        spark = SparkSession.builder.appName("HelloWorld").getOrCreate()
        pipes.log.info("Hello from AWS EMR Serverless!")
        pipes.report_asset_materialization(
            metadata={"some_metric": {"raw_value": 0, "type": "int"}},
            data_version="alpha",
        )
        spark.stop()


if __name__ == "__main__":
    main()
```

so it's clear this is running in EMR and it gives people an idea of how it fits in with spark?

Reply:
Added some actual Spark usage